app/controllers/catalog_controller.rb

```diff
@@ -61,14 +61,38 @@ def show
     super
   end
 
+  def report_issue_action(_documents)
+    mail = RecordMailer.email_report_issue(title: params[:title],
+                                           record_number: params[:record_number],
+                                           comment: params[:comment],
+                                           email: params[:email])
+    if mail.respond_to? :deliver_now
+      mail.deliver_now
+    else
+      mail.deliver
+    end
+
+    flash[:success] = t('blacklight.email.success')
+  end
+
+  def validate_report_issue_params
+    if params[:comment].blank?
+      flash[:error] = t('blackcat.report_issue.errors.comment')
+    end
+
+    flash[:error].blank?
+  end
+
```

Comment:
@awead - I don't love having this just chilling in `catalog_controller`. It seems like it's just cluttering the catalog controller and should maybe be in a separate file.

The main blacklight project puts `email_action` and `validate_email_params` in `app/controllers/concerns/blacklight/catalog.rb`. 

Should we do something similar? Or do you think it's fine to just leave this here? I defer to your Rails expertise. 🙂

Reply:
Yep. That’s a great idea. You could create a module under concerns and include it in the controller.